### ISSUE TITLE
[Feature Request] Exclude severity results

### ISSUE THREAD

**Issue by gudge25** [forcedotcom/sfdx-scanner#959]
if i run 
sfdx scanner:run --target src/main/default --category 'Best Practices,Code Style,Design,Documentation,Error,Error Prone,Stylistic Issues,Performance,Possible Errors,ECMAScript 6,Security,Variables' --severity-threshold 2 --format json

i  have  a  json file with all  severities  levels 1-5
but  i  only want to see in my son  severity  1-2 because of severity-threshold=2
if it is possible
Regards

**Comment by johnbelosf** [comment]
@gudge25 The --severity-threshold option throws an error if there problems with that severity or higher. Note that these are normalized severities from 1-3, as invoking this option also normalizes severity levels across all engines:

-s, --severity-threshold=_severity-threshold_
Throws an error when violations are found with equal or greater severity than the provided value. –normalize-severity is invoked and severity levels are reset to the baseline. Normalized severity values are: 1 (high), 2 (moderate), and 3 (low). Exit code is the most severe violation.

We currently do not have the ability to filter out rules based on severity, but we do have a number of improvements planned in how we handle and manage severity level related options. Stay tuned.

**Comment by gudge25** [comment]
@johnbelosf  yes noticed that  it shoukd be only  1-3  but for some reason  sometimes I  also see level 5  but it  is not the case and to  filter rules  is not the case
The case is  if I  have  --severity-threshold=2  and  --format json for example
And  violations are found with 2 or 1 severity (with equal or greater ) `
json also  contain severity  3  but  i better  do not have  logged severely  lover than i  expect
hope  U can understand.

**Comment by johnbelosf** [comment]
Yes I understand - as I said we are not currently filtering out violations with lower severity, the only thing that the flag does is to throw an error. So I'll keep this as a feature request.

Can you provide an example of a scan result that is throwing severity 5 violation with the --severity-threshold flag set?

**Comment by gudge25** [comment]
@johnbelosf 
{"line":"46","column":"20","endLine":"46","endColumn":"32","severity":5,"ruleName":"UnusedLocalVariable","category":"Best Practices","url":"https://pmd.github.io/pmd-6.51.0/pmd_rules_apex_bestpractices.html#unusedlocalvariable","message":"\nVariable 'casesToUpdate' defined but not used\n","normalizedSeverity":3},{"line":"53","column":"29","endLine":"53","endColumn":"30","severity":3,"ruleName":"AvoidDeeplyNestedIfStmts","category":"Design","url":"https://pmd.github.io/pmd-6.51.0/pmd_rules_apex_design.html#avoiddeeplynestedifstmts","message":"\nDeeply nested if..then statements are hard to read\n","normalizedSeverity":3},{"line":"70","column":"22","endLine":"70","endColumn":"26","severity":3,"ruleName":"ApexCRUDViolation","category":"Security","url":"https://pmd.github.io/pmd-6.51.0/pmd_rules_apex_security.html#apexcrudviolation","message":"\nValidate CRUD permission before SOQL/DML operation\n","normalizedSeverity":3},{"line":"72","column":"20","endLine":"72","endColumn":"24","severity":3,"ruleName":"AvoidDebugStatements","category":"Performance","url":"https://pmd.github.io/pmd-6.51.0/pmd_rules_apex_performance.html#avoiddebugstatements","message":"\nAvoid debug statements since they impact on performance\n","normalizedSeverity":3},

**Comment by stephen-carter-at-sf** [comment]
Code Analyzer v5 allows you to use rule selection (via `--rule-selector` flag, i.e. `-r`) to only run the rules with the severities you wish to run. For example to just run the recommended rules that are of severity 1 and 2, then do:
```
sf code-analyzer run -r Recommended:1 -r Recommended:2
```

Thus closing this issue since this is now possible with Code Analyzer v5. See: https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/code-analyzer.html
